service/qos/service_level_controller.cc

```diff
@@ -548,6 +549,25 @@ future<> service_level_controller::do_add_service_level(sstring name, service_le
     return make_ready_future();
 }
 
+service_level_map service_level_controller::get_service_levels(const bool with_static, const bool with_marked_for_deletion) const {
+    // We rely on the assumption that `_service_levels_db` (so the cache of the `service_level_controller`)
+    // is always up-to-date.
+
+    service_level_map result{};
+
+    for (const auto& [sl_name, sl] : _service_levels_db) {
+        if (!with_static && sl.is_static) {
+            continue;
+        }
+        if (!with_marked_for_deletion && sl.marked_for_deletion) {
```

Comment:
The `service_level::marked_for_deletion` field looks completely unused. I sent a PR to remove it: https://github.com/scylladb/scylladb/pull/20452

I think you can drop the `with_marked_for_deletion` parameter and completely ignore the `marked_for_deletion` field.